skada/tests/test_mapping.py

```diff
@@ -192,3 +192,54 @@ def test_new_X_adapt(estimator, da_dataset):
 )
 def test_reg_new_X_adapt(estimator, da_reg_dataset):
     _base_test_new_X_adapt(estimator, da_reg_dataset)
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    [
+        OTMappingAdapter(),
+        EntropicOTMappingAdapter(),
+        ClassRegularizerOTMappingAdapter(norm="lpl1"),
+        ClassRegularizerOTMappingAdapter(norm="l1l2"),
+        LinearOTMappingAdapter(),
+        CORALAdapter(),
+        pytest.param(
+            MMDLSConSMappingAdapter(gamma=1e-3),
+            marks=pytest.mark.skipif(not torch, reason="PyTorch not installed"),
+        ),
+    ],
+)
+def test_X_source_target_different_size(estimator):
+    # Test when X_source.shape[0] > X_target.shape[0]
+    X, y, sample_domain = make_shifted_datasets(
+        n_samples_source=20,
+        n_samples_target=10,
+        shift="covariate_shift",
+        noise=None,
+        return_dataset=False,
+    )
+
+    X_source, X_target, y_source, y_target = source_target_split(
+        X, y, sample_domain=sample_domain
+    )
+    assert X_source.shape[0] > X_target.shape[0]
+
+    # No exception should be raised
+    estimator.fit(X, y, sample_domain=sample_domain)
+
+    # Test when X_source.shape[0] < X_target.shape[0]
+    X, y, sample_domain = make_shifted_datasets(
+        n_samples_source=10,
+        n_samples_target=20,
+        shift="covariate_shift",
+        noise=None,
+        return_dataset=False,
+    )
+
+    X_source, X_target, y_source, y_target = source_target_split(
+        X, y, sample_domain=sample_domain
+    )
+    assert X_source.shape[0] < X_target.shape[0]
+
+    # No exception should be raised
+    estimator.fit(X, y, sample_domain=sample_domain)
```

Comment:
Adding new tests is maybe too much. Can you find an already existing test and change the data generation so that `X_source.shape[0] != X_target.shape[0]`?

Reply:
agreed